include/seqan/graph_types/graph_impl_directed.h

```diff
@@ -108,20 +159,34 @@ class Graph<Directed<TCargo, TSpec> >
 
 //////////////////////////////////////////////////////////////////////////////
 
-template<typename TCargo, typename TSpec>
-inline String<typename EdgeType<Graph<Directed<TCargo, TSpec> > >::Type*>&
-_getVertexString(Graph<Directed<TCargo, TSpec> > const& g) {
+template <typename TCargo, typename TSpec>
+inline String<typename EdgeType<Graph<Directed<TCargo, TSpec> > >::Type*> &
+_getVertexString(Graph<Directed<TCargo, TSpec> > const & g)
+{
     SEQAN_CHECKPOINT
     typedef Graph<Directed<TCargo, TSpec> > TGraph;
     typedef typename EdgeType<TGraph>::Type TEdgeStump;
     return const_cast<String<TEdgeStump*>&>(g.data_vertex);
 }
 
+//////////////////////////////////////////////////////////////////////////////
+
+template <typename TCargo>
+inline String<typename EdgeType<Graph<Directed<TCargo, WithSourceId> > >::Type*> &
+_getVertexInString(Graph<Directed<TCargo, WithSourceId> > const & g)
+{
+    SEQAN_CHECKPOINT
+    typedef Graph<Directed<TCargo, WithSourceId> > TGraph;
+    typedef typename EdgeType<TGraph>::Type TEdgeStump;
+    return const_cast<String<TEdgeStump*>&>(g.data_vertex_in);
```

Comment:
This is a very evil thing to do. I know the old code did not look better, but never cast-away the const type of a return value, which would allow me to literally change the state of a member of a const-object. If such thing would be intended by design, one would use the mutable keyword instead. I know this is a private function, but is there any instance where this function is called on both non-const and const graphs? If not, please return it as const reference. Otherwise add a second interface with non-const graph.